test/src/loaders/workspace_uploader/workspace_uploader_test.py

```diff
@@ -227,18 +236,74 @@ def test_post_process(setup_and_teardown):
                                "genome_upa": None,
                                "genome_filename": None}}}}
 
-    dest_file = os.path.join(
-        os.path.join(output_dir, "12345_58_1"), f"12345_58_1.fna.gz"
+    assert uploaded
+    assert expected_data == data
+
+def test_post_process_with_genome(setup_and_teardown):
+    # test with genome_tuple and genome_upa
+    params = setup_and_teardown
+    collections_source_dir = Path(params.tmp_dir) / "collections_source_dir"
+    collections_source_dir.mkdir()
+    source_dir = Path(params.tmp_dir) / "source_dir"
+    source_dir.mkdir()
+
+    host_assembly_dir2 = params.assembly_dirs[1]
+    assembly_name2 = ASSEMBLY_NAMES[1]
+    src_file2 = params.target_files[1]
+    assembly_tuple2 = workspace_uploader._WSObjTuple(
+        assembly_name2, host_assembly_dir2, "/path/to/file/in/AssembilyUtil"
+    )
+    genome_name2 = GEMOME_NAMES[1]
+    genome_tuple2 = workspace_uploader._WSObjTuple(
+        genome_name2, host_assembly_dir2, "/path/to/file/in/GenomeUtil"
+    )
+    assembly_obj_info = [4, 'name', 'type', 'time', 75, 'user', 42, 'wsname', 'md5', 78, {'foo': 'bar'}]
+    genome_obj_info = [7, 'name', 'type', 'time', 1, 'user', 42, 'wsname', 'md5', 78, {}]
+    workspace_uploader._post_process(
+        "CI",
+        88888,
+        "214",
+        collections_source_dir,
+        source_dir,
+        assembly_tuple2,
+        "42_4_75",
+        genome_tuple=genome_tuple2,
+        genome_upa="42_7_1",
+        assembly_obj_info=assembly_obj_info,
+        genome_obj_info=genome_obj_info
     )
 
+    data, uploaded = workspace_uploader._read_upload_status_yaml_file(
+        "CI", 88888, "214", host_assembly_dir2
+    )
+
+    expected_data = {
+        'CI': {88888: {'214':
+                           {'assembly_filename': assembly_name2,
+                            'assembly_upa': '42_4_75',
+                            'genome_filename': genome_name2,
+                            'genome_upa': '42_7_1'}}}}
+
     assert uploaded
     assert expected_data == data
+
+    assembly_dest_file = source_dir / "42_4_75" / "42_4_75.fna.gz"
     # check softlink
-    assert os.readlink(os.path.join(upload_dir, "12345_58_1")) == os.path.join(
-        output_dir, "12345_58_1"
+    assert os.readlink(os.path.join(collections_source_dir, "42_4_75")) == os.path.join(
+        source_dir, "42_4_75"
     )
     # check hardlink
-    assert os.path.samefile(src_file, dest_file)
+    assert os.path.samefile(src_file2, assembly_dest_file)
+
+    # check metadata file
+    metadata_file = source_dir / "42_4_75" / "42_4_75.meta"
+    with open(metadata_file, 'r') as file:
+        data = json.load(file)
+
+    assert data['upa'] == '42/4/75'
+    assert data['genome_upa'] == '42/7/1'
+    assert data['assembly_object_info'] == assembly_obj_info
+    assert data['genome_object_info'] == genome_obj_info
```

Comment:
Why not compare the entire dicts like above?

Reply:
👍

Reply:
:+1: